tests/test_intercept.js

```diff
@@ -2414,6 +2414,27 @@ test('pending mocks doesn\'t include optional mocks', function(t) {
   t.end();
 });
 
+test('calling optionally(true) on a mock makes it optional', function (t) {
+  nock('http://example.com')
+    .get('/nonexistent')
+    .optionally(true)
+    .reply(200);
+
+  t.deepEqual(nock.pendingMocks(), []);
+  t.end();
+});
+
+test('calling optionally(false) on a mock doesn\'t make it optional', function(t) {
+  nock('http://example.com')
+    .get('/nonexistent')
+    .optionally(false)
+    .reply(200);
+
+  t.notEqual(nock.pendingMocks(), []);
+  nock.cleanAll();
```

Comment:
Without this, the non-optional mock hangs around and makes the other tests fail - this is part of our cleanup.

Reply:
Might be a pitfall 🤔 

Can you add this all to the README?